atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/bucketingthings/DefaultBucketCloseTimestampCalculator.java

```diff
@@ -112,10 +120,16 @@ public OptionalLong getBucketCloseTimestamp(long startTimestamp) {
                 return OptionalLong.empty();
             }
             // start timestamp is aligned on a coarse partition
-            // MAX_BUCKET_SIZE is also a multiple of a coarse partition
+            // currentMaxBucketSize is also a multiple of a coarse partition
             // so excluding overflow, start + MAX will also be aligned with a coarse partition.
-            long cappedTimestamp =
-                    Math.min(startTimestamp + MAX_BUCKET_SIZE_FOR_NON_PUNCHER_CLOSE, freshClampedTimestamp);
+            long currentMaxCoarsePartitions = maxNumberOfCoarsePartitionsPerBucketForNonPuncherClose.get();
+            Preconditions.checkState(
+                    currentMaxCoarsePartitions > 0,
+                    "max coarse partitions must be positive",
+                    SafeArg.of("maxCoarsePartitions", currentMaxCoarsePartitions));
```

Comment:
Curious as to doing the check here as opposed to in the config object (readability wise the config object seems better / failing faster, though I get that this is maybe defensive against other rogue suppliers that might be passed in).